Parser/pgen/__main__.py

```diff
@@ -0,0 +1,51 @@
+import os
+import sys
+import argparse
+import collections
+
+from lib2to3.pgen2 import grammar, tokenize
+
+from . import token
+from . import grammar as pgen_grammar
+
+def monkey_patch_pgen2(token_lines):
+    tokens = dict(token.generate_tokens(token_lines))
+    for name, value in tokens.items():
+        setattr(tokenize, name, value)
+
+from .pgen import ParserGenerator
+
+
+def main(grammar_file, tokens_file, gramminit_h_file, gramminit_c_file, verbose):
+    with open(tokens_file) as tok_file:
+        token_lines = tok_file.readlines()
+
+    monkey_patch_pgen2(token_lines)
+
+    p = ParserGenerator(grammar_file, token_lines, verbose=verbose)
+    grammar = p.make_grammar()
+    grammar.produce_graminit_h(gramminit_h_file.write)
+    grammar.produce_graminit_c(gramminit_c_file.write)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Parser generator main program.")
```

Comment:
I would move the argument parser to the toplevel. It doesn't hurt and I like my `__main__` sections sweet and short. In fact I'd also move the `parse_args()` call into `main()`, unless you have a unittest that would greatly benefit from passing the args directly to `main()` (rather than constructing a list of command line arguments to pass to `parse_args()`).